yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -370,9 +370,11 @@ private[spark] class Client(
       try {
         hadoopConfStream.setLevel(0)
         hadoopConfFiles.foreach { case (name, file) =>
-          hadoopConfStream.putNextEntry(new ZipEntry(name))
-          Files.copy(file, hadoopConfStream)
-          hadoopConfStream.closeEntry()
+          if(file.canRead()) {
```

Comment:
If a file which was not supposed to be read by this user is not copied, it's expected behavior. I don't think we should log/flag it as a problem.